### ISSUE TITLE
API requests reached daily limits

### ISSUE THREAD

**Issue by JohanKohlstrom** [AdyRock/com.switchbot#64]
Hi, saw a similair issue in #61 - https://github.com/AdyRock/com.switchbot/issues/61

I have the Homey Switchbot app installed on two Homey Pro in two different homes. Today only using the app that in turn uses the Switchbot API.

I want to use the Switchbot API and it´s webhook capabilities but could not register a "hook" in their API due to "reached daily limits" according to their support.

The Switchbot app in Homey "talks" a lot with the API and that generates a lot of requests to the API.

Is there any workaround for this and does the app need to talk that much/often to the API?

Thanks,

I really appreciate the app!

**Comment by JohanKohlstrom** [comment]
I understand. Could a configuration in the app like "How often do you want to update the status of devices in Homey?" be an option?

Thanks for your effort!

**Comment by JohanKohlstrom** [comment]
I got a response from SwitchBot support that the daily calls to the API were 10 000.

**Comment by AdyRock** [comment]
Yes and that's at one call per device. The app calculates the update rate using 9000 / number of devices  / seconds per day. Therefore, the more devices you have the longer the interval. That means each Homey uses nearly all of the allowed calls.
One of my other apps has a setting for number of Homey's accessing the account and it multiplies the interval by that, so maybe I can add an option like that.

**Comment by AdyRock** [comment]
New test version in the store.
I have updated the webhook handling for the latest API version, which means less devices need to poll. The remaining ones that do need to poll are lower priority so I have also lengthened the period between fetches. This might help in you dual install situation.

**Comment by JohanKohlstrom** [comment]
@AdyRock, thank you for your excellent support and response time - I will try this version and see if it solves my associated issues.

**Comment by WayneLaagewaard** [comment]
Hi Johan,

ik have 6 switchbots and a hub. And every i day i see the warning that i have reached my limit. what can i do? I use a Homey Pro 2023

Wayne
